test/dispose.js

```diff
@@ -0,0 +1,42 @@
+describe('using the watcher API to dispose and watch again', function() {
+  require('./fixtures/bootstrap.js');
+  beforeEach(h.clean);
+  afterEach(h.clean);
+
+  var visible = false;
+  var element;
+  var watcher;
+
+  beforeEach(function(done) {
+    element = h.createTest({
+      style: {
+        top: '10000px'
+      }
+    });
+    h.insertTest(element);
+    watcher = inViewport(element, function() {
+      scrolled = true;
+      done();
```

Comment:
done will be called multiple times, first beforeEach never finishes since the done is in the callback